units/bson_to_parquet_backfill.go

```diff
@@ -0,0 +1,184 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// TODO (EVG-16138): Remove this job once we do the BSON to Parquet cutover.
+
+const bsonToParquetBackfillJobName = "bson-to-parquet-backfill"
+
+type bsonToParquetBackfillJob struct {
+	job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+
+	env         cedar.Environment
+	queue       amboy.Queue
+	controller  *model.BatchJobController
+	migrationID string
+}
+
+func init() {
+	registry.AddJobType(bsonToParquetBackfillJobName, func() amboy.Job { return makeBSONToParquetBackfillJob() })
+}
+
+func makeBSONToParquetBackfillJob() *bsonToParquetBackfillJob {
+	return &bsonToParquetBackfillJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    bsonToParquetBackfillJobName,
+				Version: 1,
+			},
+		},
+	}
+}
+
+// NewBSONToParquetBackfillJob returns a new Amboy job that does batched
+// migrations of existing BSON test results to Apache Parquet test results
+// stored in the Presto bucket.
+func NewBSONToParquetBackfillJob(id string) amboy.Job {
+	j := makeBSONToParquetBackfillJob()
+	j.SetID(fmt.Sprintf("%s.%s", bsonToParquetBackfillJobName, id))
```

Comment:
Does this job need scopes?

Reply:
Maybe, but not yet. The idea is that since there will be a lot if I/O wait time (reading and writing to the DB and S3) we can run these jobs in parallel across the app servers and adjust the batch size/ iterations accordingly to not overwhelm the DB.